packages/frames/src/Frame.ts

```diff
@@ -0,0 +1,186 @@
+import { MetaTag } from 'next-seo/lib/types';
+
+export enum FrameAspectRatio {
+  SQUARE = '1:1',
+  WIDE = '1.91:1',
+}
+
+export enum FrameButtonAction {
+  /**
+   * The button will trigger a post request to the postUrl.
+   * By default, the postUrl is used as the target URL if target is not provided.
+   */
+  POST = 'post',
+  /**
+   * The button will redirect to the target URL.
+   */
+  LINK = 'link',
+}
+
+export interface FrameButton {
+  /**
+   * The label to be displayed in the button.
+   */
+  label: string;
+  /**
+   * - post: The button will trigger a post request to the postUrl.
+   * - link: The button will redirect to the target URL.
+   * @default post
+   */
+  action?: FrameButtonAction;
+  /**
+   * A target URL for the button. It uses the postUrl if not provided.
+   */
+  target?: string;
+}
+
+interface FrameConstructorProps {
+  /**
+   * The title to be displayed in the frame. Required.
+   */
+  title: string;
+  /**
+   * The URL of the image to be displayed in the frame. It should be a URL. Required.
+   */
+  image: string;
+  /**
+   * The URL to be used in the post request. Required.
+   */
+  postUrl: string;
+  /**
+   * Aspect-ratio of the image. Required.
+   */
+  aspectRatio: FrameAspectRatio;
+  /**
+   * An ordered list of buttons to be displayed in the frame.
+   */
+  buttons?: FrameButton[];
+}
+
+/**
+ * Represents a Farcaster frame.
+ * Documentation: https://docs.farcaster.xyz/learn/what-is-farcaster/frames
+ * Specification: https://docs.farcaster.xyz/reference/frames/spec
+ */
+export class Frame {
+  /**
+   * The title to be displayed in the frame. Required.
+   */
+  title: string;
+  /**
+   * The URL of the image to be displayed in the frame. It should be a URL. Required.
+   */
+  image: string;
+  /**
+   * The URL to be used in the post request. Required.
+   */
+  postUrl: string;
+  /**
+   * Aspect-ratio of the image. Required.
+   */
+  aspectRatio: FrameAspectRatio;
+  /**
+   * An ordered list of buttons to be displayed in the frame.
+   */
+  buttons?: FrameButton[];
+
+  constructor(props: FrameConstructorProps) {
+    this.title = props.title;
+    this.image = props.image;
+    this.postUrl = props.postUrl;
+    this.aspectRatio = props.aspectRatio;
+    this.buttons = props.buttons;
+  }
+
+  /**
+   * Renders the frame as complete HTML markup.
+   *
+   * @example
+   * export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+   *   const frame = new Frame({ ... });
+   *   return res.status(200).send(frame.render());
+   * }
+   *
+   * @returns The HTML representation of the frame.
+   */
+  public render(): string {
+    const metaTags = this.toMetaTags().map(Frame.renderTag).join('');
```

Comment:
I suggest that the map gets expanded, because it actually pass more than argument to the function, and if tomorrow `Frame.renderTag` gets another (maybe optional) argument, it will have problems that are difficult to detect.

```suggestion
    const metaTags = this.toMetaTags().map((tag) => Frame.renderTag(tag)).join('');
```